#3180. Add test for tear-off members of JS interop type

sgrekhov opened the pull request.

eernstg reviewed (commented):
Couple of typos, awaiting further input.

srujzs reviewed (approved):
I'd also test the following:

- external extension methods on extension types
- external extension static methods on extension types
- external factories on both extension types and @staticInterop classes
- any non-externals can be torn off

sgrekhov commented:
Thank you. See below.

>- external extension methods on extension types. 

Added `interop_type_A04_t02.dart`.

>- external extension static methods on extension types. 

Added `interop_A06_t02.dart`  and  `interop_type_A04_t03.dart`.

>- external factories on both extension types and @staticInterop classes 

For extension types we already have `interop_A01_t09.dart`. For `@staticItrerop` added `staticInterop_A06_t03.dart`

> - any non-externals can be torn off

Added `interop_type_A04_t04.dart`.

PTAL.

srujzs reviewed (commented):
LGTM, thanks!
